### DISCUSSION TITLE
Custom Field of Type Subform with another Subform inside

### DISCUSSION THREAD

**Discussion by olivdee** [joomla/joomla-cms#40452] [2. Cold Features]
Hi. I am just trying out/experimenting with features, in order to push what is possible and to evaluate specific possibilities for upcoming projects. One thing that bugs me, is the missing ability to add a subform inside a subform. Why would I want to do this? Simply because I think, this way, one could replicate a kind of **"Add a custom field on the fly while editing content"-feature**. But maybe I am wrong, and even if it would be possible to have a subform inside a subform, one still couldn't achieve the AACFOTFWEC-feature... What do you think, could this be possible? Or is there maybe another workaround for this?

I will try to give an example:

- first let's call the subform inside a subform "subsubform"
- "subsubform" has the following fields: 
> -  **Field#1**: Type: "Editor", Name: "Content"
> -  **Field#2**: Type: "Media", Name: "Image"
> -  **Field#3**: Type: "List", Name: "Special Info"
- now, let's create the parent subform, and add "subsubform" as the only repeatable field
- one could then edit the content of assigned articles and select for example "Field#1", then adding via [+] another field and select "Field#3", then another "Field#3", then again a "Field#1", then a "Field#2" and so on...

Does this make sense? 
Could this work?

**Comment by olivdee** [comment]
@nikosdion Dear Nicholas, I think your solution https://github.com/nikosdion/plg_fields_inception I just stumbled upon comes really close to what I am looking for! Experimenting with it right now, but I can't find a way to select only the fields of a nested subform I need. Instead it adds all fields defined inside a nested subform... My setup is:
- new inception/nested subform field set to "repeatable"
- containing 3 different fields
- inside an article, I select the tab with the nested subform, click on "+"/add and it adds all 3 assigned fields 

= it would be great, if there is a selection letting you choose first, which one of the 3 you want to add
- is this maybe somehow already possible?
- if not, can your plugin be extended in order to behave like that?
- if yes, I would gladly donate a good amount for this feature to become reality!

**Reply by nikosdion** [reply]
That's… that's not how fields work in Joomla. Like, at all.

Maybe look into conditional fields? They're new in Joomla 4.3.

**Comment by bembelimen** [comment]
See https://github.com/joomla/joomla-cms/pull/38178 + discussion.

**Comment by olivdee** [comment]
thank you all! I was just experimenting with the Conditional Fields "Showon" feature of J!4.3, it looks very promising! But it does not work for me as described in https://github.com/joomla/joomla-cms/pull/39051#issuecomment-1519554127
